src/librustc_typeck/diagnostics.rs

```diff
@@ -1368,6 +1368,52 @@ struct Foo {
 ```
 "##,
 
+E0128: r##"
+You declared a type parameter with a default which has the same identifier as
+the type parameter. Erroneous code example:
+
+```
+pub struct Foo<SomeType=SomeType>;
+// error: type parameters with a default cannot use forward declared
+// identifiers
+```
+
+Please verify you used the good type or change the type parameter identifier.
```

Comment:
"Please verify that a name clash hasn't been accidentally introduced, and rename the type parameter if so"